modules/programs/firefox/mkFirefoxModule.nix

```diff
@@ -341,6 +340,13 @@ in {
       };
     });
 
+    profileVersion = mkOption {
+      internal = true;
+      type = types.nullOr types.ints.unsigned;
+      default = if isDarwin then null else 2;
```

Comment:
I am not sure this is correct (but I agree it will help Darwin users). `Version=2` means that `[Install]` entries are saved in `profiles.ini` and `installs.ini` is never loaded (as a backup). I think it breaks any firefox package that won't also set `MOZ_LEGACY_PROFILES=1` (for example, `firefox-bin` or a Firefox installed by `brew`.) I captured what I think is happening [here](https://github.com/nix-community/home-manager/pull/5724#issuecomment-2323655812).

---

I think the proper way to fix this is remove `Version=2` for **all** platforms (which will enforce legacy, v1 profiles behavior - the one that `firefox` `nixpkgs` package already enforces with `MOZ_LEGACY_PROFILES=1`, set for the binary wrapper).

Reply:
UPD: I stand corrected on `firefox-bin`: I think it's also [wrapped](https://github.com/NixOS/nixpkgs/blob/d80a3129b239f8ffb9015473c59b09ac585b378b/pkgs/top-level/all-packages.nix#L30124) with `wrapFirefox`, which will guarantee that `MOZ_LEGACY_PROFILES=1` is set (meaning, v1 logic is implied - using `installs.ini`.)

So the only versions of firefox affected are externally installed. This is, largely, just Darwin installations. Still, I don't think removal of `Version=2` line for all platform should affect anyone (happy to learn otherwise); so introducing a new option to set the version seems unnecessary.

Reply:
hey @brckd when you have time, do you want to weight in ?

Reply:
I just tried both wrapped and unwrapped LibreWolf versions with the following `profile.ini` with `Version = 2` removed and they seem to run fine. Perhaps the issue was with another derivative, so sorry for the confusion.
```ini
[General]
StartWithLastProfile=1

[Profile0]
Default=1
IsRelative=1
Name=Bricked
Path=bricked```